File: src/main/java/com/academy/model/FeeInvoice.java
package com.academy.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "fee_invoices")
public class FeeInvoice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long invoiceId;

    @ManyToOne
    @JoinColumn(name = "kid_id", nullable = false)
    private Kid kid;

    @ManyToOne
    @JoinColumn(name = "fee_schedule_id", nullable = false)
    private FeeSchedule feeSchedule;

    @Column(nullable = false)
    private Double amount;

    @Column(nullable = false)
    private LocalDate dueDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private InvoiceStatus status = InvoiceStatus.OPEN;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    public enum InvoiceStatus {
        OPEN, PAID, OVERDUE
    }

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: src/main/resources/football-academy-frontend/src/app/models/fee-invoice.ts
import { FeeSchedule } from "./fee-schedule";
import { InvoiceItem } from "./invoice-item";
import { Kid } from "./kid";

// export interface FeeInvoice {
//   invoiceId: number;
//   kid: Kid;
//   feeSchedule: FeeSchedule;
//   amount: number;
//   dueDate: string;
//   status: 'OPEN' | 'PAID' | 'OVERDUE';
//   createdAt: string;
//   updatedAt: string;
// }

export interface FeeInvoice {
    invoiceId: number;
    invoiceNumber: string;
    kidId?: number;
    kidName?: string;
    clientName?: string;
    amount: number;
    dueDate: string;
    status: 'OPEN' | 'PAID' | 'OVERDUE';
    createdAt: string;
    updatedAt: string;
    items: InvoiceItem[];
}
